rest-service/manager_rest/rest/rest_utils.py

```diff
@@ -751,13 +759,53 @@ def propagate_deployment_statuses(self, source):
                     )
 
             if total_srv_status or total_env_status:
-                target = self.sm.get(models.Deployment, v, locking=True)
-                if total_env_status:
-                    target.sub_environments_status = total_env_status
-                if total_srv_status:
-                    target.sub_services_status = total_srv_status
-                target.deployment_status = target.evaluate_deployment_status()
-                self.sm.update(target)
+                target = self.sm.get(
+                    models.Deployment,
+                    v,
+                    locking=True,
+                    fail_silently=True
+                )
+                if target:
+                    if total_env_status:
+                        target.sub_environments_status = total_env_status
+                    if total_srv_status:
+                        target.sub_services_status = total_srv_status
+                    target.deployment_status = \
+                        target.evaluate_deployment_status()
+                    self.sm.update(target)
+
+    def retrieve_and_display_dependencies(self, target):
+        self.create_dependencies_graph()
+        queue = [target.id]
+        visited = defaultdict(bool)
+        dependencies = []
+        while queue:
+            v = queue.pop()
+            if v not in self.graph:
+                continue
+            from_dependencies = self.sm.list(
+                models.DeploymentLabelsDependencies,
+                filters={'target_deployment_id': v}
+            )
+            if not from_dependencies:
+                continue
+            for from_dependency in from_dependencies:
+                if from_dependency.source_deployment_id in self.graph[v]:
+                    if not visited[from_dependency.source_deployment_id]:
+                        queue.append(from_dependency.source_deployment_id)
+                        dependencies.append({
+                            'parent':
+                                from_dependency.target_deployment_id,
+                            'child':
+                                from_dependency.source_deployment_id
+                        })
+                        visited[from_dependency.source_deployment_id] = True
+        dependency_display = '  [{0}] Deployment `{1}` depends on `{2}`'
+        return '\n'.join(
+            [dependency_display.format(i + 1,
+                                       d['child'],
+                                       d['parent'])
+             for i, d in enumerate(dependencies)])
```

Comment:
```suggestion
            dependency_display.format(i + 1,
                                       d['child'],
                                       d['parent'])
             for i, d in enumerate(dependencies))
```
the [] do nothing other than waste memory

Reply:
@tehasdf Done and updated